tests/tests/test-eth-fee/test-eth-paysFee.ts

```diff
@@ -0,0 +1,26 @@
+import "@moonbeam-network/api-augment";
+
+import { expect } from "chai";
+import web3 from "web3";
+
+import { describeDevMoonbeamAllEthTxTypes, DevTestContext } from "../../util/setup-dev-tests";
+import { extractInfo } from "../../util/substrate-rpc";
+import { ALITH_TRANSACTION_TEMPLATE, createTransaction } from "../../util/transactions";
+
+// We use ethers library in this test as apparently web3js's types are not fully EIP-1559
+// compliant yet.
+describeDevMoonbeamAllEthTxTypes("Ethereum - PaysFee", (context) => {
+  it("should be true for ethereum transactions", async function () {
+    const {
+      result: { events },
+    } = await context.createBlock(
+      createTransaction(context, {
+        ...ALITH_TRANSACTION_TEMPLATE,
+        value: web3.utils.toWei("1", "ether"),
+      })
+    );
+    const info = extractInfo(events);
+    expect(info).to.not.be.empty;
+    expect(info.paysFee.isYes, "Transaction should be marked as paying fees").to.be.true;
```

Comment:
Summarizing some debug about this:

I think there was previously a bug where `paysFee` was always true. You can see this in the polkadotjs UI for runtimes up until this dependency update. (Another extrinsic this impacts is `set_validation_data`.)

Now they properly return false, and this has broken some tests that relied on the buggy behavior.

This particular test should be `paysFee == false` because of https://github.com/paritytech/frontier/blob/bfc907017e9392ab702b9faddc87a34d6c683f06/frame/ethereum/src/lib.rs#L608

Reply:
We should test some other cases about the behavior when `transact` fails (in some cases it will return `pays_fee: Yes`)